api/net/Areas/Subscriber/Controllers/UserController.cs

```diff
@@ -0,0 +1,112 @@
+[ProducesResponseType(typeof(ErrorResponseModel), (int)HttpStatusCode.Forbidden)]
+public class UserController : ControllerBase
+{
+    #region Variables
+    private readonly IUserService _userService;
+    private readonly ICssHelper _cssHelper;
+    private readonly JsonSerializerOptions _serializerOptions;
+    #endregion
+
+    #region Constructors
+    /// <summary>
+    /// Creates a new instance of a UserController object, initializes with specified parameters.
+    /// </summary>
+    /// <param name="userService"></param>
+    /// <param name="serializerOptions"></param>
+    /// <param name="cssHelper"></param>
+    public UserController(IUserService userService, ICssHelper cssHelper, IOptions<JsonSerializerOptions> serializerOptions)
+    {
+        _userService = userService;
+        _cssHelper = cssHelper;
+        _serializerOptions = serializerOptions.Value;
+
+        
+    }
+    #endregion
+
+    #region Endpoints
+    /// <summary>
+    /// Find a page of user for the specified query filter.
+    /// </summary>
+    /// <returns></returns>
+    [HttpGet]
+    [Produces(MediaTypeNames.Application.Json)]
+    [ProducesResponseType(typeof(IPaged<UserModel>), (int)HttpStatusCode.OK)]
+    [SwaggerOperation(Tags = new[] { "User" })]
+    public IActionResult Find()
+    {
+        var uri = new Uri(this.Request.GetDisplayUrl());
+        var query = Microsoft.AspNetCore.WebUtilities.QueryHelpers.ParseQuery(uri.Query);
+        var result = _userService.Find(new DAL.Models.UserFilter(query));
+        var page = new Paged<UserModel>(result.Items.Select(ds => new UserModel(ds, _serializerOptions)), result.Page, result.Quantity, result.Total);
+        return new JsonResult(page);
+    }
+
+    /// <summary>
+    /// Find user for the specified 'id'.
+    /// </summary>
+    /// <param name="id"></param>
+    /// <returns></returns>
+    [HttpGet("{id}")]
+    [Produces(MediaTypeNames.Application.Json)]
+    [ProducesResponseType(typeof(UserModel), (int)HttpStatusCode.OK)]
+    [ProducesResponseType(typeof(string), (int)HttpStatusCode.NoContent)]
+    [SwaggerOperation(Tags = new[] { "User" })]
+    public IActionResult FindById(int id)
+    {
+        var result = _userService.FindById(id);
+
+        if (result == null) return new NoContentResult();
+        return new JsonResult(new UserModel(result, _serializerOptions));
+    }
+
+
+
+    /// <summary>
+    /// Update user for the specified 'id'.
+    /// Update the user in Keycloak if the 'Key' is linked.
+    /// </summary>
+    /// <param name="model"></param>
+    /// <returns></returns>
+    [HttpPut("{id}")]
+    [Produces(MediaTypeNames.Application.Json)]
+    [ProducesResponseType(typeof(UserModel), (int)HttpStatusCode.OK)]
+    [ProducesResponseType(typeof(ErrorResponseModel), (int)HttpStatusCode.BadRequest)]
+    [SwaggerOperation(Tags = new[] { "User" })]
+    public async Task<IActionResult> UpdateAsync(UserModel model)
```

Comment:
Why are we letting subscribers update users?

Reply:
Needed in order to update the user preferences when selecting my minister

Reply:
(will only ever update themselves)

Reply:
Add logic to only allow a user to update themselves.

Reply:
Will do!! Was just going to ask if that was an ok work-around :)